### ISSUE TITLE
JS built by webpacker is not being served.

### ISSUE THREAD

**Issue by kirananto** [forem/forem#9465]
**Describe the bug**

Currently, when we click on the *Write a Post* button, and start writing a new post, we are not able to publish the post.
Along with publishing the post, other actions like preview or save as draft also doesn't work. So I was just inspecting the frontend side of it and seems like there's even no event fired from the button click to begin with. FYI We just set up the config now.

**To Reproduce**

1. Go to 'Write a new Post'
2. Write something for a sample post
3. Click on 'publish'

**Expected behavior**

It should save the post

**Screenshots**

![image](https://user-images.githubusercontent.com/12209601/88281498-69190780-cd05-11ea-9cca-600acaf36403.png)

**Desktop (please complete the following information):**

- OS: Mac OS 10.15.4
- Browser: Chrome 84 & Safari 13.1
- Version: Chrome 84 & Safari 13.1

**Smartphone (please complete the following information):**

- Device:N/A
- OS:N/A
- Browser:N/A
- Version: N/A

**Additional context**

Couple of other observations and errors I've noticed is that some of the URLs are failing to load, this includes js assets.
1. /packs/js/vendor-7fea1e4befb577f5e80c.chunk.js
2. /packs/js//Search-d98cfeb3094ddbcee915.chunk.js
3. /packs/js/runtime~onboardingRedirectCheck-8e0f57a7a6bc8dba7a6a.js
4. /packs/js/1-2e718901f9ccb73f2700.chunk.js
5./packs/js/onboardingRedirectCheck-8b15c22c62c6c8c9e803.chunk.js
6. /packs/js/contentDisplayPolicy-dc03917475fc4a57885a.chunk.js
7. /packs/js/clipboardCopy-40dffcdafa2e5abc7616.chunk.js
8. /packs/js/articleForm-9e3d3e43f1ec6c2fbc40.chunk.js
9. /packs/js/vendor-7fea1e4befb577f5e80c.chunk.js

![image](https://user-images.githubusercontent.com/12209601/88281722-c8771780-cd05-11ea-9417-0cb39f3c1399.png)

It seems that some files in `/packs/js/*` is present
![image](https://user-images.githubusercontent.com/12209601/88282276-dbd6b280-cd06-11ea-8f1d-05657b40abf1.png)

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @forem/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/forem/forem/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
@kirananto I tried with Chrome 84 and it worked. Do you have any special extensions?

**Comment by kirananto** [comment]
@rhymes  Nope, no special extensions, I've tried with firefox as well, seems like it's not a browser issue, these files I've mentioned above are not built and served...

**Comment by rhymes** [comment]
@kirananto it's odd because `devto.esper.io` sounds like a proxy that doesn't belong to Forem. Do you have a system proxy activated?

**Comment by kirananto** [comment]
We're trying to set up our forum for a particular community. 
So far we've been following this guide https://docs.dev.to/installation/ for the setup.

We're facing minor issues like these, although it's set up perfectly as per the documentation. 

It'd be great if we could connect over a call, to discuss how we could join the beta program and if you guys could help us set this up, we could help contribute to the project, and help more in generalizing the codebase.

**Comment by kirananto** [comment]
FYI this happens directly when you run the docker image for forem
1. Clone code
2. Add github auth
3. docker-compose build && docker-compose up
4. Visit http://localhost:3000/new

There you'll have the issue

**Comment by kirananto** [comment]
New observation:

The hash of the files built-in logs was 
![image](https://user-images.githubusercontent.com/12209601/88445883-4987f900-ce43-11ea-96dc-2d6aae73a199.png)

But when I drop into the docker container, 
navigate to `/opt/apps/devto/public/packs/js`, The files present there is different, different hash
![image](https://user-images.githubusercontent.com/12209601/88445932-9ec40a80-ce43-11ea-88b8-e2d0a9ebb1a3.png)

**Comment by kirananto** [comment]
@rhymes When I disable the splitChunks optimization, there are no longer chunks with hashes, and the application works perfectly. So I'm assuming There's an issue with the hashing used here, and the hashing used by the webpacker library. 


```

environment.splitChunks((config) => {
  return {
    ...config,
    resolve: {
      ...config.resolve,
      alias: {
        ...(config.resolve ? config.resolve.alias : {}),
        '@crayons': path.resolve(
          __dirname,
          '../../app/javascript/crayons',
        ) /* global __dirname */,
        '@utilities': path.resolve(__dirname, '../../app/javascript/utilities'),
      },
    },
    optimization: {
      ...config.optimization,
      splitChunks: {
        ...config.optimization.splitChunks,
        cacheGroups: {
          vendor: {
            test: /node_modules/,
            chunks: 'initial',
            name: 'vendor',
            enforce: true,
          },
          default: {
            minChunks: 2,
            priority: -20,
            reuseExistingChunk: true,
          },
        },
      },
    },
  };
});
```

🤔 Wondering why this isn't affecting the dev.to platform.

**Comment by kirananto** [comment]
@joshpuetz @nickytonline   It'd be good if you could take a look at this 👍

**Comment by nickytonline** [comment]
This isn't an issue outside of docker and removing the code splitting would be detrimental to the performance of the site. In regards to hashing, it's pretty standard for webpack which is what webpacker is using under the hood. It is based on the contents of the chunked files, so the hashes are fine.

I really think this is a Docker issue or maybe caching.

Thoughts @jdoss?

**Comment by kirananto** [comment]
Yah I agree... Removing code-splitting would definitely be detrimental to the performance of the site... As this issue is not present elsewhere, I assume it's something related to the docker or caching issue...

It's pretty easy to reproduce. 
1. Clone the codebase
2. ./bin/container-setup

Once the server is up, you can see the issue. 
B/w is there any recommended docker version.

**Comment by jdoss** [comment]
@kirananto Are you running Forem in production or development mode?

**Comment by kirananto** [comment]
@jdoss I'm facing this issue in both production mode as well as development mode

**Comment by jdoss** [comment]
If you are switching back and forth between development and production you might want to try nuking your `assets` and `packs` directories and see if that helps.

As for running Forem in production mode, I wouldn't even use the `docker-compose` for anything other than development. Webpacker shouldn't be running in production mode either. If you want to play around with docker-compose in production mode, you most likely to comment out the the `webpacker` service in the `docker-compose.yml` file and then `docker-compose exec rails bundle exec rake assets:precompile` to precompile your assets and then `docker-compose restart rails` to restart everything.

We are working on a production stack for forem but it isn't ready for prime time yet.

**Comment by timorthi** [comment]
Chiming in here to report that I'm experiencing a similar (or the same?) issue using Docker on macOS - I'm getting a 404 on all the compiled JS.  


![Screen Shot 2020-08-04 at 9 32 00 PM](https://user-images.githubusercontent.com/5664347/89372360-cfedd780-d69a-11ea-901e-2106f93ac57e.png)  

The screenshot above was also for the 'Write a post' page, but the 404s occur on every page where any compiled JS assets are loaded.  

One more thing to note.. my host OS' (and therefore also my container's) `public/packs` directory contains nothing but... the manifest.json? Weird because I even got the "Compiled successfully" message from the webpacker container.


![Screen Shot 2020-08-04 at 9 47 09 PM](https://user-images.githubusercontent.com/5664347/89372859-16900180-d69c-11ea-8090-5637368092e1.png)

![Screen Shot 2020-08-04 at 9 41 47 PM](https://user-images.githubusercontent.com/5664347/89372856-15f76b00-d69c-11ea-896c-c7f10280c0ec.png)

**Comment by minhlucvan** [comment]
any updates?

**Comment by jeremyf** [comment]
@minhlucvan at present no updates.  I was unaware of this problem.  Is this still a problem?  (Note: I don't use docker for localized development).

**Comment by mirie** [comment]
We have this milestone: [Webpacker Migration](https://github.com/forem/forem/milestone/16) where we are moving away from this so closing as a won't fix.
